Sources/Confidence/EventStorage.swift

```diff
@@ -75,17 +75,20 @@ internal class EventStorageImpl: EventStorage {
             let decoder = JSONDecoder()
             let fileUrl = folderURL.appendingPathComponent(id)
             let data = try Data(contentsOf: fileUrl)
-            let dataString = String(decoding: data, as: UTF8.self)
-            return try dataString.components(separatedBy: "\n")
-                .filter { events in
-                    !events.isEmpty
-                }
-                .compactMap { eventString in
-                    guard let stringData = eventString.data(using: .utf8) else {
-                        return nil
+            if let dataString = String(data: data, encoding: .utf8) {
+                return try dataString.components(separatedBy: "\n")
+                    .filter { events in
+                        !events.isEmpty
                     }
-                    return try decoder.decode(ConfidenceEvent.self, from: stringData)
-                }
+                    .compactMap { eventString in
+                        guard let stringData = eventString.data(using: .utf8) else {
+                            return nil
+                        }
+                        return try decoder.decode(ConfidenceEvent.self, from: stringData)
+                    }
+            } else {
+                throw ConfidenceError.corruptedCache(message: "Error decoding events cache file")
```

Comment:
Is this error caught somewhere or will it crash the app?

Reply:
The error is caught upstream, so the app won't crash.

However, thanks to your comment I spent some time rethinking this part and perhaps it is better to return an empty list: the the EventSenderEngine will then try to remove these broken events from Storage [here](https://github.com/spotify/confidence-sdk-swift/blob/bbe7e4091ae0f29366a3a9dcc65b437c0f34e8c2/Sources/Confidence/EventSenderEngine.swift#L117) (added a commit for it)

Reply:
Very nice 👍